### ISSUE TITLE
Search by md5 hash

### ISSUE THREAD

**Issue by thehowl** [osuripple/cheesegull#1]

**Comment by AECX** [comment]
I would like to have something like a get parameter for md5 to get the information to this specific beatmap.
We already have:

- amount
- status
- mode
- query

Why don't we add md5?
It could help with /web/osu-osz2-getscores.php (as example) to see what AR, OD, CS, HP + StarDiff a map has. Also in what ParentID (mapset) this specific map is.

**Comment by xnyo** [comment]
Beatmap info fetching by .osu file md5 should be implemented on the API, not on the mirror.
We don't save all the information in ripple's database yet, that's why we don't have such api endpoint yet.
As for `/web/osu-osz2-getscores.php`, we currently fetch information from osu!api and we store it in db. After a while, we fetch them from the osu!api again to make sure the info is up to date.

**Comment by AECX** [comment]
Haha just letting you know I use peppy's API for now. (Thanks Nyo)
But let me know when this is going to be implemented because I'd rather use CheeseGull than poppys API.

**Comment by xnyo** [comment]
Yes, it'll be implemented, but on the API, not on cheesegull (beatmap mirror).
